src/language-js/comments.js

```diff
@@ -987,7 +987,9 @@ function getCommentChildNodes(node, options) {
   //       }
   //     }
   if (
-    (options.parser === "typescript" || options.parser === "flow") &&
+    (options.parser === "typescript" ||
+      options.parser === "flow" ||
+      options.parser === "espree") &&
```

Comment:
nit: I think it would be cleanest to just remove these lines, which does not cause any test failures for me.

(It looks like this is used for non-Babel parsers.)

Reply:
I added `espree` here because `MethodDefinition` is a estree nodeTye, in babel it's `ClassMethod`. I'll check how this function is used later.

Reply:
I am looking forward to your insights.

Reply:
@thorn0 I believe you add this function in #6400, https://github.com/prettier/prettier/pull/6400/commits/b8b81bd851ce53777dc3ba116a2427e8b346152c. Do you still remember that you add it just for safety, or you find this can't use for parsers like `__vue_expression`?

Reply:
@thorn0

Reply:
I would also be happy if we would remove this entire condition in a separate PR, rather than blocking this update as a whole.